### PR TITLE
Continue sigma delta filter

### PR DESCRIPTION
**Pull request by irish3725**
Finished SigmaDeltaFilter class. It has variables at the top that can be used to easily change the amount of images needed to initialize the background, and to buffer the background averaging. The filter method calls all of the other methods and makes all the decisions that need to be made. There is two separate actual filtering methods, but the one that isn't commented out will probably work the best.

### CODE REVIEW

File: src/test/java/csci432/SigmaDeltaFilterTest.java [hunk 1]
```diff
@@ -10,7 +10,7 @@
 import java.awt.image.BufferedImage;
 
 /**
- * Created by alex on 11/30/16.
+ * Created by alex on 12/1/16.
```
Comment (sagersmith8) [inline src/test/java/csci432/SigmaDeltaFilterTest.java:13]: LOL just remove this

File: src/main/java/csci432/SigmaDeltaFilter.java [hunk 2]
```diff
@@ -24,52 +29,89 @@ public SigmaDeltaFilter() {
...
+     * @return the newly filtered image
      **/
-    public void setBackground(BufferedImage image) {
-        if (numFiltered == 0) {
-            background = image;
-        } else {
-            refreshBackground(image);
-        }
-    }
+//    public BufferedImage filterImageRemove(BufferedImage image) {
```
Comment (sagersmith8) [inline src/main/java/csci432/SigmaDeltaFilter.java]: Use a block comment or just delete if you don't need anymore

File: src/main/java/csci432/SigmaDeltaFilter.java [hunk 3]
```diff
@@ -1,11 +1,16 @@
...
+import java.nio.Buffer;
 
 public class SigmaDeltaFilter {
 
-    private int numFiltered;
-    private BufferedImage background;
+    public int numFiltered;
+    public BufferedImage background;
+    public BufferedImage subImage;     //created separate places to save these so we can compare if we need to
+    //public BufferedImage remImage;
```
Comment (sagersmith8) [inline src/main/java/csci432/SigmaDeltaFilter.java]: Do you need this anymore?

File: src/test/java/csci432/SigmaDeltaFilterTest.java [hunk 4]
```diff
@@ -21,33 +21,94 @@ public void setup() {
...
-
-        for(int i = 0; i < 22; i++){
-            b = f.filter(a);
+    public void testRefreshBackground()  {
+        BufferedImage a = new BufferedImage(300, 300, BufferedImage.TYPE_BYTE_GRAY);
+        BufferedImage b = a;
+        BufferedImage c = a;
+        for (int i = 0; i < a.getHeight(); i++){
+            for (int j = 0; j < a.getWidth(); j++){
+                a.setRGB(j, i, 4);
```
Comment (sagersmith8) [inline src/test/java/csci432/SigmaDeltaFilterTest.java:76]: I don't think this value can possibly be right... I think it is looking for something like 255255255 for white

File: src/test/java/csci432/SigmaDeltaFilterTest.java [hunk 5]
```diff
@@ -21,33 +21,94 @@ public void setup() {
...
+                b.setRGB(j, i, 8);
```
Comment (sagersmith8) [inline src/test/java/csci432/SigmaDeltaFilterTest.java:77]: Same as above

File: src/test/java/csci432/SigmaDeltaFilterTest.java [hunk 6]
```diff
@@ -21,33 +21,94 @@ public void setup() {
...
+                c.setRGB(j, i, 6);
```
Comment (sagersmith8) [inline src/test/java/csci432/SigmaDeltaFilterTest.java:78]: Same as above

File: src/test/java/csci432/SigmaDeltaFilterTest.java [hunk 7]
```diff
@@ -21,33 +21,94 @@ public void setup() {
...
+        d = f.filterImageSubtract(b);
+        assertThat(d, equalTo(c));
+
+        //check to see if it works subtracting higher from lower
+        f.background = b;
+        d = f.filterImageSubtract(a);
+        assertThat(d, equalTo(c));
     }
 
+
```
Comment (sagersmith8) [inline src/test/java/csci432/SigmaDeltaFilterTest.java]: Delete this line

File: src/test/java/csci432/SigmaDeltaFilterTest.java [hunk 8]
```diff
@@ -21,33 +21,94 @@ public void setup() {
...
+
```
Comment (sagersmith8) [inline src/test/java/csci432/SigmaDeltaFilterTest.java]: Same as above

File: src/test/java/csci432/SigmaDeltaFilterTest.java [hunk 9]
```diff
@@ -21,33 +21,94 @@ public void setup() {
...
+
```
Comment (sagersmith8) [inline src/test/java/csci432/SigmaDeltaFilterTest.java]: Same as above

File: src/test/java/csci432/SigmaDeltaFilterTest.java [hunk 10]
```diff
@@ -21,33 +21,94 @@ public void setup() {
...
+
```
Comment (sagersmith8) [inline src/test/java/csci432/SigmaDeltaFilterTest.java]: Same as above

File: src/test/java/csci432/SigmaDeltaFilterTest.java [hunk 11]
```diff
@@ -21,33 +21,94 @@ public void setup() {
     }
 
     @Test
-    public void testRefreshBackground()  {
-        f.refreshBackground(new BufferedImage(300, 300, BufferedImage.TYPE_BYTE_BINARY));
-    }
-
-    @Test
-    public void testSetBackground() {
+    public void testFilter() {
```
Comment (sagersmith8) [inline src/test/java/csci432/SigmaDeltaFilterTest.java:24]: Sweet looks like the filter works like you expected! Thats awesome!

File: src/test/java/csci432/SigmaDeltaFilterTest.java (hunk 4 above)
Comment (irish3725) [reply src/test/java/csci432/SigmaDeltaFilterTest.java:76]: These small numbers were just used to see if the method would do the subtraction was actually subtracting in each image.

**Pull request merged**